### PR TITLE
[log] add log to thread attach function

### PR DESCRIPTION
**Pull request by zhanglongxia**

### CODE REVIEW

File: src/agent/thread_helper.cpp [hunk 1]
```diff
@@ -255,9 +255,18 @@ void ThreadHelper::Attach(const std::string &         aNetworkName,
...
+    otbrLog(OTBR_LOG_INFO,
+            "Thread Attach, NetworkName:%s, PanId:0x%04x, ExtPanId:0x%02x%02x%02x%02x%02x%02x%02x%02x, Channel:%u, "
+            "MasterKey: Hidden, PSKc: Hidden",
+            aNetworkName.c_str(), aPanId, extPanId.m8[0], extPanId.m8[1], extPanId.m8[2], extPanId.m8[3],
+            extPanId.m8[4], extPanId.m8[5], extPanId.m8[6], extPanId.m8[7], channel);
+
 exit:
     if (error != OT_ERROR_NONE)
     {
+        otbrLog(OTBR_LOG_WARNING, "Thread Attach Failed");
```
Comment (simonlingoogle) [inline src/agent/thread_helper.cpp:268]: ERROR instead of WARNING ?

File: src/agent/thread_helper.cpp [hunk 2]
```diff
@@ -255,9 +255,18 @@ void ThreadHelper::Attach(const std::string &         aNetworkName,
     SuccessOrExit(error = otThreadSetPskc(mInstance, &pskc));
 
     SuccessOrExit(error = otThreadSetEnabled(mInstance, true));
+
+    otbrLog(OTBR_LOG_INFO,
+            "Thread Attach, NetworkName:%s, PanId:0x%04x, ExtPanId:0x%02x%02x%02x%02x%02x%02x%02x%02x, Channel:%u, "
+            "MasterKey: Hidden, PSKc: Hidden",
+            aNetworkName.c_str(), aPanId, extPanId.m8[0], extPanId.m8[1], extPanId.m8[2], extPanId.m8[3],
```
Comment (simonlingoogle) [inline src/agent/thread_helper.cpp:262]: Use some method like: `extPanId.ToString()` ?

File: src/agent/thread_helper.cpp (hunk 1 above)
Comment (bukepo) [reply src/agent/thread_helper.cpp:268]:
```suggestion
        otbrLog(OTBR_LOG_WARNING, "Failed to attach to Thread network: %s", otThreadErrorToString(error));
```

File: src/agent/thread_helper.cpp [hunk 3]
```diff
@@ -255,9 +255,18 @@ void ThreadHelper::Attach(const std::string &         aNetworkName,
     SuccessOrExit(error = otThreadSetPskc(mInstance, &pskc));
 
     SuccessOrExit(error = otThreadSetEnabled(mInstance, true));
+
+    otbrLog(OTBR_LOG_INFO,
+            "Thread Attach, NetworkName:%s, PanId:0x%04x, ExtPanId:0x%02x%02x%02x%02x%02x%02x%02x%02x, Channel:%u, "
+            "MasterKey: Hidden, PSKc: Hidden",
```
Comment (bukepo) [inline src/agent/thread_helper.cpp:261]:
```suggestion
    otbrLog(OTBR_LOG_INFO,
            "Attaching to Thread network, NetworkName:%s, PanId:0x%04x, ExtPanId:0x%02x%02x%02x%02x%02x%02x%02x%02x, Channel:%u, "
            "MasterKey: Hidden, PSKc: Hidden",
```

File: src/agent/thread_helper.cpp (hunk 1 above)
Comment (zhanglongxia) [reply src/agent/thread_helper.cpp]: Updated

File: src/agent/thread_helper.cpp (hunk 3 above)
Comment (zhanglongxia) [reply src/agent/thread_helper.cpp]: Updated

File: src/agent/thread_helper.cpp (hunk 2 above)
Comment (wgtdkp) [reply src/agent/thread_helper.cpp:262]: > Use some method like: `extPanId.ToString()` ?

I would recommend
```c++
std::string BytesToHexString(const uint8_t *aBuf, size_t aBufLength);
```

Comment (zhanglongxia) [reply src/agent/thread_helper.cpp:262]: Only this place prints the extended panid. There is no need to define a function to convert the panid to string.

**Comment by jwhui** [comment]
Closing stale PR.

**Pull request closed without merging**